src/main/java/org/sonarlint/intellij/SonarLintIntelliJClient.kt

```diff
@@ -177,6 +174,15 @@ object SonarLintIntelliJClient : SonarLintClient {
         showBalloon(null, params.text, convert(params.type))
     }
 
+    override fun showSmartNotification(params: ShowSmartNotificationParams) {
+        val projects = params.scopeIds.stream()
+            .map { BackendService.findModule(it)?.project ?: BackendService.findProject(it) }
+            .filter { it != null}
```

Comment:
You can combine map and filter with `mapNotNull`

Reply:
Thanks, I didn't know!